cdxev/auxiliary/filename_gen.py

```diff
@@ -0,0 +1,134 @@
+import logging
+import re
+import typing as t
+import unicodedata
+from datetime import datetime, timezone
+
+from dateutil.parser import isoparse
+
+logger = logging.getLogger(__name__)
+
+
+def generate_filename(sbom: dict) -> str:
+    """
+    Generates a filename for the given SBOM.
+
+    If the SBOM doesn't contain any of the required metadata to generate a unique filename,
+    it will default to ``bom.json``.
+
+    :param dict sbom: The SBOM to generate a filename for.
+
+    :return: The filename.
+    """
+    name = sbom.get("metadata", {}).get("component", {}).get("name", "")
+    name = _sanitize(name)
+    version = sbom.get("metadata", {}).get("component", {}).get("version", "")
+    version = _sanitize(version)
+    timestamp_str: t.Union[str, None] = sbom.get("metadata", {}).get("timestamp")
+
+    if not name and not version and not timestamp_str:
+        return "bom.json"
+
+    try:
+        timestamp = isoparse(timestamp_str)  # type: ignore # because type errors are caught below
+    except (ValueError, TypeError):
+        logger.info(
+            "SBOM has no or an unparsable timestamp. Using current time in filename."
+        )
+        timestamp = datetime.now(timezone.utc)
+
+    name = name or "unknown"
+    timestamp_str = _timestamp_to_utc_str(timestamp)
+
+    components = [name]
+    if version:
+        components.append(version)
+
+    components.append(timestamp_str)
+
+    return "_".join(components) + ".cdx.json"
+
+
+def generate_validation_pattern(sbom: dict) -> str:
+    """
+    Creates a regular expression which can be used to validate the filename of the given SBOM.
+
+    The pattern allows the following variants of the filename:
+
+    * ``bom.json`` is always an allowed name.
+    * ``<name>_[<version>_][<hash>_<timestamp>|<hash>|<timestamp>].cdx.json``, where
+      * ``<name>`` == ``metadata.component.name``, if it exists, otherwise ``unknown``.
+      * ``<version>`` MUST be present, if and only if ``metadata.component.version`` exists.
+      * ``<hash>`` is ``metadata.component.hashes[x].content`` for
```

Comment:
for what?

Reply:
That's what happens when you go on lunch break mid-sentence 😆 
Fixed in 25f41ea